src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunctionComparer.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.PowerFx.Core.Functions
+{
+    internal class TexlFunctionComparer : IEqualityComparer<TexlFunction>
+    {
+        public bool Equals(TexlFunction x, TexlFunction y)
+        {
+            return object.Equals(x, y);
```

Comment:
>return object.Equals(x, y); [](http://example.com/codeflow?start=12&length=27)

I think this is the best we can do. 
But worth adding a comment to explain why. 

Comparing params is temping, but can be wrong since params are often useless (blame CheckInvocation). 

For example, some functions like "If" don't even have params.